scripts/fast-reboot

```diff
@@ -557,6 +559,29 @@ function reboot_pre_check()
 
     check_db_integrity
 
+    # Check SSD health
+    if [ -x "${SSD_UTIL}" ]; then
+        debug "Checking ssd health before ${REBOOT_TYPE}..."
+        health_line=$(${SSD_UTIL} | grep -E "Health\s*:\s*[0-9]+\.?[0-9]*%" || true)
+
+        if [ -z "$health_line" ]; then
+            debug "Warning: Health line not found in ${SSD_UTIL} output."
+        else
+            # Extract the numeric value X from "Health       : X%"
+            health_value=$(echo "$health_line" | sed -n 's/[^0-9]*\([0-9]\+\).*/\1/p')
+
+            # Check if health_value is a valid number
+            if [ -z "$health_value" ]; then
+                debug "Warning: Could not find health percentage."
+            elif [ "$health_value" -gt 0 ]; then
+                debug "SSD Health is $health_value% — OK."
+            else
+                error "Warning: Health is $health_value% — Possible drive failure!"
+                exit "${EXIT_SDD_HEALTH_FAILURE}"
```

Comment:
From Boyang - this is very basic / minimal check that can be done to prevent the issue. 

The core idea  is to collect debug information to then implement a solution to really prevent these issues.